### PR TITLE
Merge branch 'merge_stable_convert' into merge_stable

### PR DESCRIPTION
**Pull request by MartinNowak**
Conflicts:
    src/access.d
    src/aggregate.d
    src/aliasthis.d
    src/apply.d
    src/argtypes.d
    src/arrayop.d
    src/arraytypes.d
    src/attrib.d
    src/builtin.d
    src/canthrow.d
    src/clone.d
    src/cond.d
    src/constfold.d
    src/cppmangle.d
    src/ctfeexpr.d
    src/dcast.d
    src/dclass.d
    src/declaration.d
    src/delegatize.d
    src/denum.d
    src/dimport.d
    src/dinterpret.d
    src/dmacro.d
    src/dmangle.d
    src/dmodule.d
    src/doc.d
    src/dscope.d
    src/dstruct.d
    src/dsymbol.d
    src/dtemplate.d
    src/dversion.d
    src/errors.d
    src/escape.d
    src/expression.d
    src/func.d
    src/globals.d
    src/hdrgen.d
    src/identifier.d
    src/init.d
    src/inline.d
    src/json.d
    src/lexer.d
    src/lib.d
    src/link.d
    src/mars.d
    src/mtype.d
    src/nogc.d
    src/nspace.d
    src/objc.d
    src/opover.d
    src/optimize.d
    src/parse.d
    src/root/file.d
    src/root/man.d
    src/root/outbuffer.d
    src/sapply.d
    src/sideeffect.d
    src/statement.d
    src/staticassert.d
    src/target.d
    src/tokens.d
    src/traits.d
    src/visitor.d

### CODE REVIEW

File: src/mtype.d [hunk 1]
```diff
@@ -250,7 +250,8 @@ extern (C++) Type stripDefaultArgs(Type t)
                             for (size_t j = 0; j < params.dim; j++)
                                 (*params)[j] = (*parameters)[j];
                         }
-                        (*params)[i] = new Parameter(p.storageClass, ta, null, null);
+                        StorageClass stc = p.storageClass & (~STCauto); // issue 14656
+                        (*params)[i] = new Parameter(stc, ta, null, null);
```
Comment (9rnsr) [inline src/mtype.d]: This part is incorrect.

Comment (MartinNowak) [inline src/mtype.d]: Why do we have 2 different fixes in master and stable?
https://github.com/D-Programming-Language/dmd/pull/5032
https://github.com/D-Programming-Language/dmd/pull/4728

Comment (9rnsr) [inline src/mtype.d]: It had introduced by #4728. And later it's changed again back to the original code by #4729. On the other hand, current stable branch contains only #4728, so this change is actually reverting part of #4729.

#5032 is not directly related.

Comment (MartinNowak) [inline src/mtype.d]: Fixed

Comment (9rnsr) [inline src/mtype.d]: thanks.

**Comment by 9rnsr** [comment]
Auto-merge toggled on

**Pull request merged**